### PR TITLE
Ignore deleted projects

### PR DESCRIPTION
**Pull request by carise**

### CODE REVIEW

File: src/main/java/com/google/cloud/security/scanner/sources/LiveProjectSource.java [hunk 1]
```diff
@@ -223,7 +223,10 @@ private boolean refreshProjects(String nextPageToken) throws IOException {
         if (project.getParent() != null) {
           orgId = project.getParent().getId();
         }
-        this.projects.add(new GCPProject(project.getProjectId(), orgId, project.getName()));
+        if (project.getLifecycleState() == null
```
Comment (blueandgold) [inline src/main/java/com/google/cloud/security/scanner/sources/LiveProjectSource.java]: Just want to double-check here that the correct condition here is null, vs NOT null?

Comment (blueandgold) [inline src/main/java/com/google/cloud/security/scanner/sources/LiveProjectSource.java]: nit: purely optional, for multiple conditions, maybe break it up for easier readability, and is flexible in the future to tack on more conditions.

if (project.getLifecycleState() == null) {
  continue;
}
if (project.getLifecycleState().startsWith("DELETE")) {
  continue;
}
this.projects.add(new GCPProject(project.getProjectId(), orgId, project.getName());

**Comment by carise** [comment]
@blueandgold PTAL, made some updates on the conditions and also the unit tests

**Comment by blueandgold** [comment]
👍

**Pull request merged**